src/schemas/components/SchemaFormPropertyAnyOf.vue

```diff
@@ -53,16 +57,34 @@
     'update:value': [SchemaValue],
   }>()
 
-  const value = computed({
-    get() {
-      return propertyValues[selectedPropertyIndex.value]
-    },
-    set(value) {
-      propertyValues[selectedPropertyIndex.value] = value
-      emit('update:value', value)
-    },
+  const internalValue = computed(() => {
+    if (isPrefectKindValue(props.value)) {
+      return props.value
+    }
+
+    return propertyValues[selectedPropertyIndex.value]
   })
 
+  async function updateValue(value: unknown): Promise<void> {
+    // if its a prefect kind value like workspace_variable or json we don't need to do anything
+    if (isPrefectKindValue(value)) {
+      emit('update:value', value)
+      return
+    }
+
+    let index = await getInitialIndexForSchemaPropertyAnyOfValue({ value, property: props.property, api, schema })
+
+    if (index === -1) {
+      console.error('Could not determine property index')
+      index = selectedPropertyIndex.value
+    }
+
+    propertyValues[index] = value
+    selectedPropertyIndexValue.value = index
+
+    emit('update:value', value)
+  }
```

Comment:
This mostly makes sure that the correct `anyOf` is shown when resetting to the default value. Also fixes an issue where if you were using json it would inadvertently update the `anyOf` index that was last selected.